src/software/ai/passing/pass_generator_test.cpp

```diff
@@ -61,7 +61,8 @@ class PassGeneratorTest : public testing::Test
                 break;
             }
         }
-        EXPECT_LE(std::abs(curr_score - prev_score), min_score_diff);
+        // TODO: uncomment this once we know why this fails
+        // EXPECT_LE(std::abs(curr_score - prev_score), min_score_diff);
```

Comment:
@garethellis0 / @MathewMacDougall can you take a look at this?

Reply:
Ok here are the results from my testing (for both this and the commented out line below)

Only the `check_pass_converges` test failes, despite having equally tight tolerances on convergence as other tests.

**Info on the failure in the `waitForConvergence` function:**
When it fails (in this PR), the `curr_score=0.799159` and `prev_score=0.858256`

When it passes on master, the `curr_score=0.0.843504` and `prev_score=0.844879`

Looking at the sequence of score values, it it still converging to the right place (~0.844) just the numbers are slightly less stable than before which is why convergence is not detected. Even increasing the number of iterations from 30 to 40 does not fix the issue, because it is "converged" already.

**Info on the failure in the `check_pass_converged` function, specifically checking the pass speeds are the same**
Similar to the above. We hvae converged to the right pass but are slightly more unstable. At one iteration, the score drops by `~0.06` due to changes in receiver position, time, and most notably pass speed. Running for more iterations shows we converge back to the right score / pass speed (which is fine most of the time).

**History of changes to this test**
There are no other recent changes that are suspicious. When making the `passGenerator` deterministic, it was actually made easier to pass the convergence test because fewer values had to be within tolerance in a row.
During the `angleSweep` changes, the tolerance value slightly increased.

**My conclusion**
This PR has not broken anything. Evan's `anglesweep` changes did seem to introduce a bit more instability (he posted a scatterplot), and that may just be exacerbated here due to slight numerical differences between compilers (IIRC we had similar issues with test tolerances when switching to clang). It should be fine to increase the tolerances as needed to made this pass, but I also think it's worth filing a ticket to investigate the stability of our convergence. Likely all these minor changes added up, and gradient descent is already not perfectly stable and may just need to be re-tuned.

**My personal conclusion**

Reply:
Here are the plots for reference
![image (4)](https://user-images.githubusercontent.com/21165407
/73014138-801a8380-3dce-11ea-89a8-11911f668030.png)
![image (3)](https://user-images.githubusercontent.com/21165407/73014139-801a8380-3dce-11ea-865e-310c4f2ba1fa.png)

Reply:
they mysteriously started passing again on my machine, but I'll wait for CI